docs/installation/autoware/docker-installation.md

```diff
@@ -16,21 +16,137 @@
 
-!!!warning
+Install Docker Engine and its post-installation steps:
+
+```bash
+# Taken from: https://docs.docker.com/engine/install/ubuntu/
+# And: https://docs.docker.com/engine/install/linux-postinstall/
+
+# Uninstall old versions
+sudo apt-get remove docker docker-engine docker.io containerd runc
+
+# Install using the repository
+sudo apt-get update
+
+sudo apt-get install \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+
+# Add Docker’s official GPG key:
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+
+# Install Docker Engine
+sudo apt-get update
+sudo apt-get install docker-ce docker-ce-cli containerd.io
 
-    Be very careful with this method. Make sure you read and confirmed all the steps in the Ansible configuration before using it.
+# Verify that Docker Engine is installed correctly by running the hello-world image.
+sudo docker run hello-world
+# Note: This command downloads a test image and runs it in a container. When the container runs, it prints a message and exits.
 
-If you want to manually install the dependencies:
+# Post-installation steps for Linux
+
+# Create the docker group.
+sudo groupadd docker
+
+# Add your user to the docker group.
+sudo usermod -aG docker $USER
+
+# Log out and log back in so that your group membership is re-evaluated.
+
+# Verify that you can run docker commands without sudo
+docker run hello-world
+# Note: This command downloads a test image and runs it in a container. When the container runs, it prints a message and exits.
+```
+
+Install Nvidia Container Toolkit:
+
+```bash
+# Taken from https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#setting-up-nvidia-container-toolkit
+
+# Setup the package repository and the GPG key:
+distribution=$(. /etc/os-release;echo $ID$VERSION_ID) \
+      && curl -s -L https://nvidia.github.io/libnvidia-container/gpgkey | sudo apt-key add - \
+      && curl -s -L https://nvidia.github.io/libnvidia-container/$distribution/libnvidia-container.list | sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
+
+# Install the nvidia-docker2 package (and dependencies) after updating the package listing:
+sudo apt-get update
+sudo apt-get install -y nvidia-docker2
+
+# Restart the Docker daemon to complete the installation after setting the default runtime:
+sudo systemctl restart docker
+
+# At this point, a working setup can be tested by running a base CUDA container:
+sudo docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi
+
+# This should result in a console output shown below:
+# +-----------------------------------------------------------------------------+
+# | NVIDIA-SMI 450.51.06    Driver Version: 450.51.06    CUDA Version: 11.0     |
+# |-------------------------------+----------------------+----------------------+
+# | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
```

Comment:
@kenji-miyake @xmfcx is there a way to make `spell-check-differential` ignore sections of a file? I don't think `Disp` and `Uncorr` should be added to the `cspell` dictionary just to avoid this warning.

Reply:
I don't know but since this is an optional CI check, we can ignore it too.

Reply:
Good question! Yes, there are several ways.
1. Use `cspell:ignore word1, word2`.
2. Use `cspell:disable-line`.
3. Use `cspell:disable` and `cspell:enable`.
4. Add `ignoreRegExpList` in `.cspell.json`.

![image](https://user-images.githubusercontent.com/31987104/161021213-d6f15d5a-2270-44b9-bb09-6a3bd07686b1.png)

https://github.com/tier4/autoware-spell-check-dict/blob/08ab79ce04b5af3ca0aff00d6cca1a905b15b7b4/.cspell.json#L43

In this case, I guess writing the No. 1 at the top of this file would be clean.
![image](https://user-images.githubusercontent.com/31987104/161021590-0757ea50-3c34-4183-83fa-a7839103420d.png)

@esteve How do you feel?

Reply:
https://github.com/autowarefoundation/autoware/pull/141/files#diff-41f1e40341b8d84975bb0de78a9fd14351611575fc472827065247bd41116246R65

Added it to here, thanks!